adapters/acuityads/acuityads.go

```diff
@@ -0,0 +1,193 @@
+package acuityads
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"text/template"
+
+	"github.com/mxmCherry/openrtb"
+	"github.com/prebid/prebid-server/adapters"
+	"github.com/prebid/prebid-server/errortypes"
+	"github.com/prebid/prebid-server/macros"
+	"github.com/prebid/prebid-server/openrtb_ext"
+)
+
+type AcuityAdsAdapter struct {
+	endpoint template.Template
+}
+
+func NewAcuityAdsBidder(endpointTemplate string) *AcuityAdsAdapter {
+	template, err := template.New("endpointTemplate").Parse(endpointTemplate)
+	if err != nil {
+		return nil
+	}
+	return &AcuityAdsAdapter{endpoint: *template}
+}
+
+func GetHeaders(request *openrtb.BidRequest) *http.Header {
+	headers := http.Header{}
+	headers.Add("Content-Type", "application/json;charset=utf-8")
+	headers.Add("Accept", "application/json")
+	headers.Add("X-Openrtb-Version", "2.5")
+
+	if request.Device != nil {
+		if len(request.Device.UA) > 0 {
+			headers.Add("User-Agent", request.Device.UA)
+		}
+
+		if len(request.Device.IPv6) > 0 {
+			headers.Add("X-Forwarded-For", request.Device.IPv6)
+		}
+
+		if len(request.Device.IP) > 0 {
+			headers.Add("X-Forwarded-For", request.Device.IP)
+		}
+	}
+
+	return &headers
+}
+
+func (a *AcuityAdsAdapter) MakeRequests(
+	openRTBRequest *openrtb.BidRequest,
+	reqInfo *adapters.ExtraRequestInfo,
+) (
+	requestsToBidder []*adapters.RequestData,
+	errs []error,
+) {
+
+	var errors []error
+	var acuityAdsExt *openrtb_ext.ExtAcuityAds
+	var err error
+
+	for i, imp := range openRTBRequest.Imp {
+		acuityAdsExt, err = a.getImpressionExt(&imp)
```

Comment:
It seems that we are rewritting `acuityAdsExt *openrtb_ext.ExtAcuityAds` over and over again ignoring the first n-1 `ext` values.
```
[]Imp={ {...Ext:`values`}, {...Ext:`otherValues`}, {...Ext:`lastExtValues`} }
   ignored--------^--------------------^
```
Unless every `Imp` element in the array come with the exact same `ext` to which we could call `a.getImpressionExt(&imp)` only once:
```
51   func (a *AcuityAdsAdapter) MakeRequests(
52       openRTBRequest *openrtb.BidRequest,
53       reqInfo *adapters.ExtraRequestInfo,
54   ) (
55       requestsToBidder []*adapters.RequestData,
56       errs []error,
57   ) {
58  
59       var errors []error
60 -     var acuityAdsExt *openrtb_ext.ExtAcuityAds
61 -     var err error
   +     acuityAdsExt, err := a.getImpressionExt(&openRTBRequest.Imp[0])
   +     if err != nil {
   +         errors = append(errors, err)
   +     }
62  
63       for i, imp := range openRTBRequest.Imp {
64 -         acuityAdsExt, err = a.getImpressionExt(&imp)
65 -         if err != nil {
66 -             errors = append(errors, err)
67 -             break
68 -         }
69           openRTBRequest.Imp[i].Ext = nil
70       }
71  
72 -     if len(errors) > 0 {
73 -         return nil, errors
74 -     }
75       
76       url, err := a.buildEndpointURL(acuityAdsExt)
77       if err != nil {
78           return nil, []error{err}
79       }
adapters/acuityads/acuityads.go
```